packages/attune/src/server/repo/dist/delete.rs

```diff
@@ -85,10 +86,13 @@ pub async fn handler(
     .await
     .unwrap();
 
-    // If no rows were affected, the distribution was already deleted.
-    // We return an OK response here because the user's goal was accomplished regardless.
+    // If no rows were affected, the distribution was already deleted or never existed.
     if result.rows_affected() == 0 {
-        return Ok(Json(DeleteDistributionResponse::default()));
+        return Err(ErrorResponse::builder()
+            .status(axum::http::StatusCode::NOT_FOUND)
+            .error("REPO_NOT_FOUND")
+            .message("repository not found")
+            .build());
```

Comment:
[eh whatever] What I meant was "you should `SELECT` for the repository rather than inferring that the repository doesn't exist", in case the application has a bug that violates invariants.

Reply:
I want to make sure we're on the same page- do you mean something different than what we're already doing up here? https://github.com/attunehq/attune/blob/beed778e51de0fbd8dc9dee04cf1f914b2a61749/packages/attune/src/server/repo/dist/delete.rs#L36-L55

It seems to me that:
- L36-L55 covers "what if the repository doesn't exist".
- L77-L96 should therefore only result in a 404 if if the repo exists but the distribution doesn't.
- I don't check for the repo+distribution combo first because why query twice when once will do.

I did update the error message and code to disambiguate this case though, realized I missed that the first time through.

Does that match with what you'd ideally do?